### PR TITLE
Add batched inference possibility

### PR DESCRIPTION
**Pull request by alonsosilvaallende**
By providing a clone attribute to the logits processors they get copied for all the batch:
I found it here in the vllm code (not documented I think):
https://github.com/vllm-project/vllm/blob/19dcc02a72e3ed52e3bf95aae44ea1f40ce42ea0/vllm/sampling_params.py#L537-L550

Solves #14

### CODE REVIEW

**Comment by alonsosilvaallende** [comment]
Thank you very much @aerdem4 
I have fixed the issues. Let me know if something else is needed.

**Comment by alonsosilvaallende** [comment]
I tried to do what you mentioned but I don't think I succeded.

**Comment by aerdem4** [comment]
Now it has a merge conflict. You may get help from an LLM to make it work.

**Comment by alonsosilvaallende** [comment]
I tried with the help of claude. Does it work now?

**Review by aerdem4 — APPROVED** [review]

**Comment by aerdem4** [comment]
Thank you. I just published the new version on pypi.
